SimPPS/PPSPixelDigiProducer/src/RPixChargeShare.cc

```diff
@@ -0,0 +1,157 @@
+#include "SimPPS/PPSPixelDigiProducer/interface/RPixChargeShare.h"
+#include <iostream>
+#include <fstream>
+
+RPixChargeShare::RPixChargeShare(const edm::ParameterSet &params, uint32_t det_id)
+  : det_id_(det_id), theRPixDetTopology_(), sqrt_2(sqrt(2.0))
+{
+  verbosity_ = params.getParameter<int>("RPixVerbosity");
+  signalCoupling_.clear();
+  ChargeMapFile2E_[0] = params.getParameter<std::string>("ChargeMapFile2E");
+  ChargeMapFile2E_[1] = params.getParameter<std::string>("ChargeMapFile2E_2X");
+  ChargeMapFile2E_[2] = params.getParameter<std::string>("ChargeMapFile2E_2Y");
+  ChargeMapFile2E_[3] = params.getParameter<std::string>("ChargeMapFile2E_2X2Y");
+  
+  double coupling_constant_ = params.getParameter<double>("RPixCoupling");
+
+  signalCoupling_.push_back(coupling_constant_);
+  signalCoupling_.push_back( (1.0-coupling_constant_)/2 );
+  
+  no_of_pixels_ = theRPixDetTopology_.detPixelNo();
+
+  double xMap, yMap;
+  double chargeprobcollect;
+  int xUpper[]={75,150,75,150};
+  int yUpper[]={50,50,100,100};
+  int ix,iy;
+  for(int i=0;i<4;i++){
+   std::ifstream fChargeMap (ChargeMapFile2E_[i]);
+   if (fChargeMap.is_open())
+   {
+     while (fChargeMap>>xMap>>yMap>>chargeprobcollect )
+     {
+      ix = int((xMap+xUpper[i])/5);
+      iy = int((yMap+yUpper[i])/5);
+      chargeMap2E[i][ix][iy]=chargeprobcollect;     
+     }
+     fChargeMap.close();
+   }
+   else throw cms::Exception("RPixChargeShare") << "Charge map file not found"; 
+  }
+}
+
+std::map<unsigned short, double, std::less<unsigned short> >  RPixChargeShare::Share(
+     const std::vector<RPixSignalPoint> &charge_map)
+{
+  thePixelChargeMap.clear();
```

Comment:
`thePixelChargeMap` seems to be used only in this function, in which case it would be better to be a local variable. Then the `return` will make use of NRVO to avoid a copy of the map.